### PR TITLE
Tk 0 add new cryptocurrency

### PR DESCRIPTION
**Pull request by thom-as-thom**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
-import Home from './src/components/Home';
+import Home from './src/screens/Home';
+import {NavigationContainer} from '@react-navigation/native';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {AddCurrency} from './src/screens/AddCrypto';
+import store from './src/store';
+
+const Stack = createNativeStackNavigator();
 
 const App = () => {
```
Comment (tomasdematos) [inline App.tsx:11]: use implicit return 

const app = () => (
<Provider....
)

File: App.tsx [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
-  return <Home />;
+  return (
+    <Provider store={store}>
+      <NavigationContainer>
+        <Stack.Navigator screenOptions={{headerShown: false}}>
+          <Stack.Screen name="home" component={Home} />
```
Comment (tomasdematos) [inline App.tsx:16]: you should create a file routes, 
src/navigation/routes. where you store an enum of the routes:

export enum routes{
  HOME = 'HOME',
}

File: src/components/Button/index.tsx [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 
-import {Title, Touchable} from './styles';
+import {RegularTitle, Title, Touchable, YellowTouchable} from './styles';
 
-const Button: FC<Props> = ({text}): ReactElement => (
-  <Touchable>
-    <Title> {text} </Title>
-  </Touchable>
-);
+const Button: FC<Props> = ({text, onClick, type, disabled}) => {
```
Comment (tomasdematos) [inline src/components/Button/index.tsx:7]: use implicit return

File: src/components/Button/index.tsx [hunk 4]
```diff
@@ -1,12 +1,21 @@
-import React, {FC, ReactElement} from 'react';
+import React, {FC} from 'react';
 import {Props} from '../../types';
```
Comment (tomasdematos) [inline src/components/Button/index.tsx:2]: the Props should be declare in the same file of the component

File: src/components/CryptoView/index.tsx [hunk 5]
```diff
@@ -13,48 +13,63 @@ import {
...
-
-      <Change>
-        {coin.metrics.market_data.percent_change_usd_last_24_hours > 0 ? (
-          <MaterialIcon name="call-made" size={18} color={theme.colors.green} />
-        ) : (
-          <MaterialIcon
-            name="call-received"
-            size={18}
-            color={theme.colors.red}
+const Cryptoview = ({coin}: {coin: CoinData}): JSX.Element => {
```
Comment (tomasdematos) [inline src/components/CryptoView/index.tsx:19]: declare and use Props

File: src/components/CryptoView/index.tsx [hunk 6]
```diff
@@ -13,48 +13,63 @@ import {
...
+  const dispatch = useDispatch();
+  const Cryptos: CoinData = useSelector(
+    state => state.addedCryptos.addedCryptos,
+  );
+  const DeleteCrypto = () => {
```
Comment (tomasdematos) [inline src/components/CryptoView/index.tsx:24]: rename, deleteCrypto

File: src/components/ListContainer/index.tsx [hunk 7]
```diff
@@ -1,41 +1,34 @@
...
-  }, [Cryptos]);
-
+const ListContainer = (): JSX.Element => {
+  const Cryptos = useSelector(state => state.addedCryptos.addedCryptos);
+  const navigation = useNavigation();
   const renderItem = ({item}) => <Cryptoview coin={item} />;
-
+  const onClick = () => {
+    navigation.navigate('AddCurrency');
+  };
```
Comment (tomasdematos) [inline src/components/ListContainer/index.tsx:15]: let a blanck line before the return

File: src/components/TinyImage/index.tsx [hunk 8]
```diff
@@ -1,8 +1,8 @@
-import React, {FC, ReactElement} from 'react';
+import React, {FC} from 'react';
 import {TinyLogo} from './styles';
 import {Props} from '../../types';
 
-const TinyImage: FC<Props> = ({imgUrl}): ReactElement => (
+const TinyImage: FC<Props> = ({imgUrl}) => (
```
Comment (tomasdematos) [inline src/components/TinyImage/index.tsx:5]: declare props in this file, for this component

File: src/screens/AddCrypto/index.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+import {useDispatch, useSelector} from 'react-redux';
+import {useState} from 'react';
+import {addCrypto} from '../../store/actions';
+import theme from '../../theme';
+
+export const AddCurrency = (): JSX.Element => {
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const {addedCryptos} = useSelector(state => state.addedCryptos);
+  const [Search, setSearch] = useState('');
```
Comment (tomasdematos) [inline src/screens/AddCrypto/index.tsx:14]: rename, Search -> searc, Focus -> focus

File: src/components/CryptoView/index.tsx [hunk 10]
```diff
@@ -13,48 +13,63 @@ import {
...
+    dispatch(deleteCrypto(coin.Asset.id, Cryptos));
+  };
```
Comment (VicenBayugar) [inline src/components/CryptoView/index.tsx:26]: leave a blank space before de return

File: src/store/actions.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+};
+
+export const deleteCrypto: Function = (Key, Cryptos) => {
+  const NewList = Cryptos.filter(item => item.Asset.id !== Key);
+  console.log(Cryptos);
```
Comment (VicenBayugar) [inline src/store/actions.ts:31]: delete this log

**Pull request merged**